centreon/src/Core/Migration/Infrastructure/Repository/DbReadMigrationRepository.php

```diff
@@ -0,0 +1,149 @@
+use Core\Migration\Application\Repository\MigrationInterface;
+use Core\Migration\Application\Repository\ReadMigrationRepositoryInterface;
+use Core\Migration\Domain\Model\ExecutedMigration;
+use Core\Migration\Domain\Model\NewMigration;
+
+class DbReadMigrationRepository extends AbstractRepositoryRDB implements ReadMigrationRepositoryInterface
+{
+    use LoggerTrait;
+
+    /** @var MigrationInterface[] */
+    private $migrations;
+
+    /**
+     * @param DatabaseConnection $db
+     * @param \Traversable<MigrationInterface> $migrations
+     */
+    public function __construct(
+        DatabaseConnection $db,
+        \Traversable $migrations,
+    ) {
+        $this->db = $db;
+
+        if (iterator_count($migrations) === 0) {
+            throw new \Exception('Migrations not found');
+        }
+
+        $this->migrations = iterator_to_array($migrations);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function findAvailableMigrations(): array
+    {
+        $migrations = [];
+
+        foreach ($this->migrations as $migration) {
+            $shortName = (new \ReflectionClass($migration))->getShortName();
+
+            $migrations[] = new NewMigration(
+                $shortName,
+                $migration->getModuleName(),
+                $migration->getDescription(),
+            );
+        }
+
+        return $migrations;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function findExecutedMigrations(): array
+    {
+        $result = $this->db->query($this->translateDbName('SHOW TABLES FROM `:db` LIKE "migrations"'));
+        if (! $result || $result->rowCount() === 0) {
+            $this->notice('Migrations table does not exist yet, considering no migrations has been done.');
+
+            return [];
+        }
+
+        $query = $this->translateDbName(
+            <<<'SQL'
+                SELECT SQL_CALC_FOUND_ROWS m.id, m.name, mi.name as module_name, m.executed_at
+                FROM `:db`.migrations m
+                LEFT JOIN modules_informations mi ON mi.id = m.module_id
+                SQL
+        );
+
+        $statement = $this->db->query($query);
+
+        if (! $statement) {
+            return [];
+        }
+
+        $result = $statement->fetchAll(\PDO::FETCH_ASSOC);
+
+        $migrations = [];
+        foreach ($result as $migrationData) {
+            $migrations[] = new ExecutedMigration(
+                $migrationData['name'],
+                $migrationData['module_name'] ?: ExecutedMigration::CORE_MODULE_NAME,
+                $migrationData['id'],
+                (new \DateTime())->setTimestamp($migrationData['executed_at']),
+            );
+        }
+
+        return $migrations;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function findNewMigrations(): array
+    {
+        $availableMigrations = $this->findAvailableMigrations();
+
+        $executedMigrations = $this->findExecutedMigrations();
```

Comment:
Did those methods will be use outside of the class later ? If no they could be declare as private